api/types.proto

```diff
@@ -447,13 +447,25 @@ message AcceptancePolicy {
 }
 
 
+message ExternalCA {
+	enum CAType {
+		CFSSL = 0 [(gogoproto.enumvalue_customname) = "CATypeCFSSL"];
+	}
+
+	// CAType is the type of this external CA.
+	CAType type = 1;
+
+	// URL is the URL where the external CA can be reached.
+	string url = 2 [(gogoproto.customname) = "URL"];
```

Comment:
@jefferai: My main goal in this PR is to refactor the external CA aspects of the API so that we can add things like this in a backward-compatible way. I don't think there's any urgency in adding these fields because we can easily add them in future PRs that will make use of them. But I think there is some urgency in changing "external CAs" from a list of strings to a list of structs by Docker 1.12-rc3, so that we don't force a backward-incompatible API change.

But I'm happy to add HTTP headers and/or CA options right now if we're confident we know what form they should take in the API.